src/miner.cpp

```diff
@@ -104,13 +104,6 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
     if (Params().MineBlocksOnDemand())
         pblock->nVersion = GetArg("-blockversion", pblock->nVersion);
 
-    // Create coinbase tx
-    CMutableTransaction txNew;
-    txNew.vin.resize(1);
-    txNew.vin[0].prevout.SetNull();
-    txNew.vout.resize(1);
-    txNew.vout[0].scriptPubKey = scriptPubKeyIn;
-
     // Add dummy coinbase tx as first transaction
```

Comment:
The old code (and the new code) is just reserving a placeholder transaction for the coinbase. It isn't fully constructed yet, and must be replaced later, so it is irrelevant where it is constructed. (It *should* probably be constructed in the scope that I move it to.)

I think your question was worth asking though.